server/models/db/workerTimeslot.js

```diff
@@ -21,7 +21,19 @@ async function updateWorkerAvailability(workerId, specificDate, newStatus) {
     return db.any(`update ${Tables.workerTimeslot} set status='${newStatus}' where date='${specificDate}' and worker_id='${workerId}';`);
 }
 
+async function checkWorkerAvailability(workerTimeslotId) {
+    //Selects all from workerTimeslot table 
+    const queryStatement = `select * from ${Tables.workerTimeslot}  where worker_timeslot_id = ${workerTimeslotId}`;
+    const queryOutput = await db.any(queryStatement);
+    //check if workertimeSlot is present 
+    if (_.isEmpty(queryOutput)) return false;
+    //check if available 
+    const currentStatus = _.map(queryOutput, 'status');
+    if (currentStatus == 'available') return true;
```

Comment:
Hi Nancy,
I am confused by this line of code. Based on line 31, it looks like currentStatus is an array of status values. Here, we are checking if an array is equal to a string which would cause issues. Instead, I would recommend using lodash's `includes` to check if the array includes an available status. Alternatively, during the mapping in line 31, you could also check if the status value is equal to `available` and return true if any value is equal to true. Else, return false at the very end.

Reply:
Hi Amy,

I would like to clarify this comment with you. Currently, the code does the following:

1. The query selects all attributes from the workerTimeslot table where worker_timeslot_id = `workerTimeslotId`. The output of this query is stored in the constant  called `queryOutput`
This is an intermediary step to ensure the `checkWorkerAvailability` function is pulling the correct entries from the database. (Line 27)
2. We check if the `queryOutput` is empty, aka no such entry exists in the workerTimeslot table> Return false. (Line 29)
3. If there is a valid entry corresponding to the specified `workerTimeslotId`,  we store the value of the `status` parameter as the constant  `currentStatus`. (Line 31).
4. If the`currentStatus` is available, then we return true. (Line 32).
5. Else, if the `currentStatus` is unavailable, then we return false at the end. (Line 33).

Could you please clarify what you'd like me to change?
Thanks!

Reply:
HI Nancy,
Yep, I follow your logic. Can you console log what the currentStatus is? I get the value of an array, e.g., ['available'] or ['unavailable'] which would result in the condition on line 32 to always be false.

Reply:
Hi Amy,
Thank you for clarifying!
My Action: I have replaced the `const currentStatus = _.includes(queryOutput, 'status');`  with `const currentStatus = _.includes(arr,'available');`

You can see this change in line 31 of server/routes/private.js

Please confirm if this takes care of your suggestion!